river/Output.zig

```diff
@@ -105,10 +105,18 @@ pub fn init(self: *Self, wlr_output: *wlr.Output) !void {
     // refresh rate), and each monitor supports only a specific set of modes. We
     // just pick the monitor's preferred mode, a more sophisticated compositor
     // would let the user configure it.
-    if (wlr_output.preferredMode()) |mode| {
-        wlr_output.setMode(mode);
+    if (wlr_output.preferredMode()) |preferred_mode| {
+        wlr_output.setMode(preferred_mode);
         wlr_output.enable(true);
-        try wlr_output.commit();
+        wlr_output.commit() catch |err| if (!wlr_output.modes.empty()) {
+            var it = wlr_output.modes.iterator(.forward);
+            while (it.next()) |mode| {
+                if (mode == preferred_mode) continue;
+                wlr_output.setMode(mode);
+                wlr_output.commit() catch continue;
+                break;
+            } else return err;
+        } else return err;
```

Comment:
Is this if really necessary? I think this could be simplified to the following
```suggestion
    wlr_output.commit() catch |err| {
        var it = wlr_output.modes.iterator(.forward);
        while (it.next()) |mode| {
            if (mode == preferred_mode) continue;
            wlr_output.setMode(mode);
            wlr_output.commit() catch continue;
            break;
        } else {
            return err;
        }
    };
```

Reply:
Ah, right. Blindly copied from sway 🤦